docs/guide/authjs/session-data.md

```diff
@@ -0,0 +1,118 @@
+# Session data
+
+This guide explains how to add custom data to the user session. In many cases, you may wish to adapt which information is returned by the authentication flow. This can depend on your provider or any additional API calls you may make to enrich the session data.
+
+## Modify the JWT Token
+
+In order to persist data between session requests, we need to inject certain information into the JWT token, which we can then access during subsequent session requests. However, avoid injecting too much data into the JWT token, as it is limited in size. We recommend only injecting an access or a session token, that can then be used to request further user information inside the session callback.
+
+The JWT callback provides:
+- `token`: The raw JWT token
+- `account`, `profile`, `isNewUser`: The data returned by the OAuth provider. This data varies by provider, we recommend logging each value to inspect what data is included. These values are only avalible on creation of the JWT token, in subsequent requests only the `token` will be accessible. 
+
+
+```ts
+import { NuxtAuthHandler } from '#auth'
+
+export default NuxtAuthHandler({
+  // your authentication configuration here!
+  callbacks: {
+    jwt({ token, account, profile }) {
+        if (account) {
+            token.sessionToken = account.session_token
+        }
+        return token
+    },
+  }
+})
+```
+
+:::info
+Any data injected into the JWT token, cannot be directly accessed from the frontend, however can be accessed on the server side using the `getToken` function.
+:::
+
+## Inject data into the Session
+
+After enriching the JWT token with additional data, you can now access this data inside the `session` callback. The `session` callback is made everytime that the session data is requested. This can happen when using `useAuth`, `getServerSideSession` or when the session is refreshed. 
```

Comment:
```suggestion
After enriching the JWT token with additional data, you can now access this data inside the `session` callback. The `session` callback is invoked every time the session data is requested. This can happen when using `useAuth`, `getServerSideSession` or when the session is refreshed. 
```